src/chainstate/burn/db/sortdb.rs

```diff
@@ -2448,32 +2436,30 @@ impl <'a> SortitionHandleTx <'a> {
                 //   update the reward set
                 if let Some(reward_info) = recipient_info {
                     let mut current_len = self.get_reward_set_size()?;
-                    let mut recipient_indexes: Vec<_> = reward_info.recipients.iter().map(|(_, x)| *x).collect();
+                    let (_, recipient_index) = reward_info.recipient;
                     let mut remapped_entries = HashMap::new();
                     // sort in decrementing order
-                    recipient_indexes.sort_unstable_by(|a, b| b.cmp(a));
-                    for index in recipient_indexes.into_iter() {
-                        // sanity check
-                        if index >= current_len {
-                            unreachable!("Supplied index should never be greater than recipient set size");
-                        } else if index + 1 == current_len {
-                            // selected index is the last element: no need to swap, just decrement len
-                            current_len -= 1;
-                        } else {
-                            let replacement = current_len - 1; // if current_len were 0, we would already have panicked.
-                            let replace_with =
-                                if let Some((_prior_ix, replace_with)) = remapped_entries.remove_entry(&replacement) {
-                                    // the entry to swap in was itself swapped, so let's use the new value instead
-                                    replace_with
-                                } else {
-                                    self.get_reward_set_entry(replacement)?
-                                };
-
-                            // swap and decrement to remove from set
-                            remapped_entries.insert(index, replace_with);
-                            current_len -= 1;
-                        }
+
```

Comment:
Why was the sort removed?  Is it guaranteed that these items are deterministically ordered still?

Reply:
The sort was only used to set the order of the recipients being removed from the reward set -- now there's only one recipient being removed, so a sort isn't required (the dangling comment line there should be deleted though).

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1863/commits/9238aa1032bbb2cea8f5fc1366e7d912b35cbdb7